### PR TITLE
Added the opc-comment-input and opc-comment-list component.

### PR DESCRIPTION
**Pull request by gisumit**
# Added the opc-comment-input and opc-comment-list component.

#### Does this PR introduce a breaking change
No

## Screenshots

<details>
<summary>View Screenshots</summary>

Default view:
Screenshot 2020-09-22 at 2 43 49 PM

and, Adding the text to comment input it activates the comment button.
Screenshot 2020-09-29 at 11 58 50 AM

comment list component:
Screenshot 2020-09-29 at 11 41 22 AM

</details>

### Ready-for-merge Checklist

- [X] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [X] Does the change have appropriate unit tests?
- [X] Did tests pass?
- [X] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [X] My changes generate no new warnings
- [X] I have performed a self-review of my own code
- [X] I have documented my code, particularly in areas like todo, complex logic, quick fix, temporary patch, etc.
#### If it is a new component
- [X] Does your component follow One Platform style guidelines?
- [X] Does your component web accessibility standards? [Helper Doc](https://www.w3.org/TR/wai-aria-1.1/)
- [X] Does your component support desktop screen sizes (350px, 720px, 1150px ,1920px)
#### Browsers you have tested in
- [X] Latest two versions of Mozilla Firefox and Google Chrome supported by Red Hat Enterprise Linux distribution
- [X] Google Chrome [Latest 2 versions]
- [X] Mozilla Firefox [Latest 2 versions]
- [ ] Microsoft Edge [Latest 2 versions]
- [ ] Apple Safari [Latest 2 versions]
- [ ] Microsoft Internet Explorer 11

### CODE REVIEW

File: packages/opc-comment-input/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+![Version](https://img.shields.io/badge/version-0.0.1-blue.svg?cacheSeconds=2592000)
+![Build Status](https://travis-ci.org/dwyl/esta.svg?branch=master)
+[![Maintenance](https://img.shields.io/badge/Maintained%3F-yes-green.svg)](https://github.com/1-Platform/op-components/graphs/commit-activity)
+
+A standardized web component based on Lit Element for Red Hat One Platform to use as a comment input component.
+
+## Prerequisites
+The `opc-comment-input` component is dependent on the Patternfly library for styling. So to avoid any missing styles, add the `patternfly.css` file before the component script tags.
+
+# Usage
```
Comment (hybridx) [inline packages/opc-comment-input/README.md:12]: @gisumit Can you use the newer format which has examples for Angular and React

File: packages/opc-comment-input/jest.config.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+    preset: 'ts-jest',
+    runner: 'jest-electron/runner',
+    testEnvironment: 'jest-electron/environment',
+    setupFiles: ['./dist/opc-comment-input.js'],
+};
```
Comment (hybridx) [inline packages/opc-comment-input/jest.config.js:6]:
```suggestion
};

```

**Review by hybridx — CHANGES_REQUESTED** [review]

**Review by ghanlohar — APPROVED** [review]
Looks good to me.

**Review by sayak-sarkar — APPROVED** [review]

**Review by hybridx — APPROVED** [review]

File: packages/opc-comment-input/declaration.d.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+declare module '*.scss';
+// Add your type definitions here
+
+type OPCComment = {
+  name?: string;
```
Comment (diwanshi) [inline packages/opc-comment-input/declaration.d.ts:5]: name field doesnt imply that its being used for commentor. rename `name` to `commentor`

File: packages/opc-comment-input/src/opc-comment-input.scss [hunk 4]
```diff
@@ -0,0 +1,35 @@
+@import '@one-platform/opc-styles';
+
+:host {
+  display: block;
+}
+
+.comment--empty__state {
```
Comment (diwanshi) [inline packages/opc-comment-input/src/opc-comment-input.scss:7]: these style names are not following the proper bem structure.

File: packages/opc-comment-input/src/opc-comment-input.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+          </button>
+          <button class="pf-c-button pf-m-primary"
+            ?disabled="${!this.commentText.length}"
+            @click="${(e: HTMLElementEventMap) => this._commentButtonClicked()}">Comment</button>
+        </div>
+      </div>
+    `;
+  }
+
+  _commentButtonClicked() {
```
Comment (diwanshi) [inline packages/opc-comment-input/src/opc-comment-input.ts:56]: move these functions above render function.

File: packages/opc-comment-input/src/opc-comment-input.ts [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  static get styles() {
+    return [ style ];
+  }
+
+  render() {
+    return html`
+      ${ !this.comments.length ?
+        html`
+          <div class="comment--empty__state">
+            <p><strong>No comments found...</strong> but you can be the first to start the conversations.</p>
```
Comment (diwanshi) [inline packages/opc-comment-input/src/opc-comment-input.ts:87]: rephrase this sentence to `No comments yet! Type your comment to start the conversation`

**Review by diwanshi — CHANGES_REQUESTED** [review]

File: packages/opc-comment-input/README.md (hunk 1 above)
Comment (gisumit) [reply packages/opc-comment-input/README.md:12]: yeah, added usage for angular, react, and vanilla js in the README.md file.

Comment (gisumit) [reply packages/opc-comment-input/README.md:12]: can you review it again?
@diwanshi

**Review by diwanshi — DISMISSED** [review]
Please use the latest format for readme and your bem struvture in css doesn't look right to me. Please take a reference from timeline component or opc-header component

**Pull request merged**